hueplusplus/include/HueDeviceTypes.hpp

```diff
@@ -0,0 +1,123 @@
+/**
+    \file DeviceTypes.h
+    Copyright Notice\n
+    Copyright (C) 2020  Jan Rogall      - developer\n
+    Copyright (C) 2020  Moritz Wirger   - developer\n
+
+    This file is part of hueplusplus.
+
+    hueplusplus is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    hueplusplus is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with hueplusplus.  If not, see <http://www.gnu.org/licenses/>.
+**/
+
+
+#ifndef _HUEDEVICETYPES_H
+#define _HUEDEVICETYPES_H
+
+#include <set>
+#include <string>
+
+#include "HueExceptionMacro.h"
+#include "HueLight.h"
+
+static const std::set<std::string> c_EXTENDEDCOLORLIGHT_GAMUTB_TYPES = {
```

Comment:
Consider moving strings into cpp file, so adding types requires fewer recompliling.
Also, they could be static variables in a function, so there are no issues with static initialization order.
Example:
```cpp
const std::set<std::string>& getXXXTypes() {
    static const std::set<std::string> XXXTypes = {...};
    return XXXTypes;
}
```